plugins/modules/domain.py

```diff
@@ -62,9 +73,17 @@ def run_module():
         value=dict(required=True),
         name=dict(required=True),
         domain=dict(required=True),
-        record_type=dict(choices=['A', 'CNAME', 'TXT', 'AAAA', 'NS', 'SRV', 'MX', 'DKIM', 'SPF'], default='A'),
-        state=dict(choices=['present', 'absent'], default='present')
-    ))
+        record_type=dict(choices=['A', 'AAAA', 'CAA',
+                                  'CNAME', 'DKIM', 'DMARC',
+                                  'DNAME', 'LOC', 'MX',
+                                  'NAPTR', 'NS', 'PTR',
+                                  'SPF', 'SRV', 'SSHFP',
+                                  'TLSA', 'TXT'],
+                         default='A'),
+        state=dict(choices=['present', 'absent'], default='present'),
+        record_ttl=dict(type='int',required=False, default=0)
```

Comment:
The default should be null or empty string to let the API handle the default.

Reply:
Do not agree, the API does not explictly specify the default it uses ( which eventually is 0 when you use the API, and 600 if you use OVH API). 
I think specifying an explicit default value is far better than implicit default value that might change.

Reply:
I don't have a strong advice on this topic.
"0"' means "use the zone TTL", and I think this is a good default